envelope.go

```diff
@@ -199,6 +199,31 @@ func parseBinaryOnlyBody(root *Part, e *Envelope) error {
 	return nil
 }
 
+func parseTextBody(p *Part) ([]byte, error) {
+
+	// use reader that part described
+	allBytes, ioerr := ioutil.ReadAll(p)
+	if ioerr != nil {
+
+		// quoted-printable was not quoted. try raw
+		if strings.Contains(ioerr.Error(), "quotedprintable: invalid unescaped byte") {
+			reader, err := newCharsetReader("utf-8", p.rawReader)
```

Comment:
I may be wrong, but my understanding is that at this point, `p.rawReader` has already been partially or completely consumed.  The plain text portion of the message is lost.  Your unit test below seems to confirm that; I will continue comment there.